src/fluree/db/util/context.cljc

```diff
@@ -0,0 +1,105 @@
+(ns fluree.db.util.context
+  (:require [clojure.string :as str]))
+
+;; handles some default context merging.
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(defn mapify-context
+  "An unparsed context may be a vector of maps or URLs. This merges them together into one large map.
+
+  If a sequence context is used, and an item is an empty string, will substitute a default context."
+  [context default-context]
+  (when context
+    (cond
+
+      (map? context)
+      context
+
+      (sequential? context)
+      (reduce (fn [acc context-item]
+                (cond
+
+                  (map? context-item)
+                  (merge acc context-item)
+
+                  (= "" context-item)
```

Comment:
If the syntax `["", {"ex": "http://example.org/ns/"}]` is json-ld standard, is there a reason to not put this functionality in the json-ld lib and use that? At first blush I do feel a bit weird about implementing this in a `util`s file on db, rather than in our standard lib.

Reply:
The merging syntax is standard, but here we are injecting the ledger's default context when it is referenced.

I used `""` as a shortcut to the ledger it is sitting on (itself). You should be able to name other ledgers to use their context, but I don't see that as high priority.

The JSON-LD library doesn't make external calls, so would need to think about how to put it there where it can't call a Fluree ledger to get a context. To solve this for other contexts (e.g. schema.org), we embedded the context int he the JSON-LD library itself. We can't do that for an evolving Fluree ledger. Even this is a temporary solution though, not a long-term one.

Reply:
Oh, so it's not the `""` that's standard, but rather the vector syntax, and we have chosen `""` to mean "default"?

Reply:
> Oh, so it's not the `""` that's standard, but rather the vector syntax, and we have chosen `""` to mean "default"?

Yes, you can provide just a string for context, but that is treated as a URL which you much perform a "get" to retrieve the actual context. Not sure what the URL would be to a specific Fluree ledger.

JSON-LD spec has a few places they allow an empty string `""` to indicate the 'base' or parent IRI, so I think the approach here is still in the spirit of the spec. (see: https://www.w3.org/TR/json-ld11/#base-iri)